MDGameSynchronizer/MDGameSynchronizer.cs

```diff
@@ -328,38 +329,45 @@ public void SendPlayerPingEvent(int PeerId, int Ping)
 
         private void OnPlayerJoinedEvent(int PeerId)
         {
-            // Check if this is our own join message
-            if (PeerId == MDStatics.GetPeerId())
+            // Check if this is our own join message or if we are a client
+            if (PeerId == MDStatics.GetPeerId() || MDStatics.IsClient())
             {
                 return;
             }
+            
+            MDGameSynchPeerInfo PeerInfo = new MDGameSynchPeerInfo(this, PeerId);
+            PeerSynchInfo.Add(PeerId, PeerInfo);
 
-            // Check if we are a client and leave
-            if (MDStatics.IsClient())
+            if (IsPauseOnJoin())
             {
-                return;
+                PauseGame();
+                foreach (int peerid in GameSession.GetAllPeerIds().Where(peerid => peerid != MDStatics.GetServerId()))
```

Comment:
Couldn't you extend the condition to include `&& peerid != PeerId` ?

Reply:
Also, is the case where someone leaves mid-synch handled to unpause the game?

Reply:
Yes the MDGameSynchPeerInfo is removed in the OnPlayerLeave event, and the check synch complete timer is checking this. However now this timer is not started on join but rather on synch start. I will move it so it is started on join.